jdaviz/configs/imviz/plugins/footprints/footprints.py

```diff
@@ -431,12 +434,25 @@ def _on_is_active_changed(self, *args):
         if self.is_active:
             self._ensure_first_overlay()
 
-        for overlay, viewer_marks in self.marks.items():
-            for viewer_id, marks in viewer_marks.items():
-                visible = self._mark_visible(viewer_id, overlay)
+        selected_obj = self.viewer.selected_obj
+        if selected_obj is None:
+            return
+
+        viewers = self.viewer.selected_obj if self.viewer.is_multiselect else [self.viewer.selected_obj]    # noqa E501

```

Comment:
nit [if-refactoring]: use `selected_obj` here instead of `self.viewer.selected_obj`
```suggestion
        viewers = selected_obj if self.viewer.is_multiselect else [selected_obj]
```